plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -99,47 +86,62 @@ def main():
             application_id=dict(required=False),
             changelog=dict(required=False),
             description=dict(required=False),
-            revision=dict(required=False),
+            revision=dict(required=True),
             user=dict(required=False),
-            appname=dict(required=False),
-            environment=dict(required=False),
-            validate_certs=dict(default=True, type='bool'),
         ),
-        required_one_of=[['app_name', 'application_id']],
-        supports_check_mode=True
+        required_one_of=[['app_name', 'application_id']]
```

Comment:
```suggestion
        required_one_of=[['app_name', 'application_id']],
        mutually_exclusive=[['app_name', 'application_id']],
```
Then you can remove the manual check below.